coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetViewHelperServiceImpl.java

```diff
@@ -247,4 +248,24 @@ public ProposalBudgetService getProposalBudgetService() {
 	public void setProposalBudgetService(ProposalBudgetService proposalBudgetService) {
 		this.proposalBudgetService = proposalBudgetService;
 	}
+
+    public boolean displaySubawardPeriodWarning(Budget budget) {
+        return hasMultiplePeriods(budget) && !isBudgetLineItemExists(budget);
+    }
+
+
+    private boolean hasMultiplePeriods(Budget budget) {
+        return budget.getBudgetPeriods().size() > 1;
+    }
+
+    private boolean isBudgetLineItemExists(Budget budget) {
+        boolean lineItemExists = false;
+        for(BudgetPeriod budgetPeriod : budget.getBudgetPeriods()) {
+            if(budgetPeriod.getBudgetPeriod() > 1 && budgetPeriod.getBudgetLineItems().size() > 0) {
+                lineItemExists = true;
+                break;
```

Comment:
So if line items exist in any but the first period you don't need the warning? The warning seems to say it needs line items in all periods?